app/services/users/omniauth.rb

```diff
@@ -0,0 +1,35 @@
+class Users::Omniauth
+  def initialize(auth)
+    @auth = auth
+  end
+
+  def user
+    find_user_by_uid || update_user || create_user
+  end
+
+  private
+
+  attr_reader :auth
+
+  def find_user_by_uid
+    User.find_by(provider: auth.provider, uid: auth.uid)
+  end
+
+  def find_user_by_email
```

Comment:
Cool 👍. I suggest two things. First, let's make `find_user_by_email` a lazy-property. There is no reason to make DB request twice. Second, let's slide it down because we usually read code from top to bottom.
```ruby
def update_user_by_email
  user_by_email.update(provider: auth.provider, uid: auth.uid) if user_by_email.present?
end

def user_by_email
  unless defined? @user_by_email
    @user_by_email = User.find_by(email: auth.info.email)
  end
  @user_by_email
end
```

Reply:
@atatarenko Because of reek warning, I will change this this way:

```ruby
def user_by_email
   @user_by_email ||= User.find_by(email: auth.info.email)
end
```